meson.build

```diff
@@ -4,7 +4,7 @@ project('harfbuzz', 'c', 'cpp',
   default_options: [
     'cpp_eh=none',          # Just to support msvc, we are passing -fno-exceptions also anyway
     'cpp_rtti=false',       # Just to support msvc, we are passing -fno-rtti also anyway
-    'cpp_std=c++11',
+    'cpp_std=c++14',        # Just to support msvc, their STL requires it
```

Comment:
Would this enforce c++14 on all compilers, or just MS compilers? We definitely haven't decided yet to set a C++14 minimum requirement.

Reply:
Yes, that's correct. Unfortunately that's already required by your MSVC support; see https://learn.microsoft.com/en-us/cpp/build/reference/std-specify-language-standard-version?view=msvc-170#remarks.

Reply:
It's fine if MSVC requires it. But we still support older gcc/clang that don't support C++14 as far as I know.

Reply:
Is it possibly to conditionally do that?

Reply:
Not through `default_options` as I understand it. You can try using manual `add_project_arguments` but removing the default_options entry will instead open Harfbuzz up to https://github.com/mesonbuild/meson/issues/12202 (due to the lower minimum dependency).

Reply:
Meson upstream does something like this: https://github.com/mesonbuild/meson/blob/master/manual%20tests/4%20standalone%20binaries/meson.build#L5-L7

In this case, global arguments have higher precedence so we could use that. I'll send it to you now.